Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs

```diff
@@ -36,11 +36,19 @@ public static IWebDriver ClickndWait(this IWebDriver driver, By by, TimeSpan tim
 
         public static void Click(this IWebElement element, bool ignoreStaleElementException = true)
         {
+            Click(element, null, ignoreStaleElementException);
+        }
+        public static void Click(this IWebElement element, IWebDriver driver, bool ignoreStaleElementException = true)
+        {
+            //Use this Click if you would like to wait until the click is complete before returning.  WaitForTransaction() is UCI specific.
             try
             {
                 element.Click();
+
+                if (driver != null)
+                    driver.WaitForTransaction();
```

Comment:
This is going to impact all Click Events. Even for the Web Client and the WaitForTransaction may not be valid.   We need to move this into the API that is making the click.